datafusion/core/src/physical_plan/windows/mod.rs

```diff
@@ -90,12 +93,26 @@ pub fn create_window_expr(
             order_by,
             window_frame,
         )),
-        WindowFunction::AggregateUDF(fun) => Arc::new(PlainAggregateWindowExpr::new(
-            udaf::create_aggregate_expr(fun.as_ref(), args, input_schema, name)?,
-            partition_by,
-            order_by,
-            window_frame,
-        )),
+        WindowFunction::AggregateUDF(fun) => {
```

Comment:
The major difference with this PR and what @stuartcarnie did is that this PR still allows AggregateUDFs to run as window functions even if they don't implement `retract_batch`

Reply:
Update: it turns out it is not correct to run AggregateUDFs as sliding window function if they don't implement `retract_batch` which I have made clear in comments and the tests